### PR TITLE
APIGW: Fix tests to run with non-default account ID

### PR DESCRIPTION
**Pull request by viren-nadkarni**
## Summary

This PR greenifies following tests when using a non-default account ID and/or region.

- [ ]  tests.aws.services.apigateway.test_apigateway_basic.TestAPIGateway
    - [ ]  test_api_gateway_lambda_proxy_integration
    - [ ]  test_api_gateway_lambda_proxy_integration_with_path_param
    - [ ]  test_api_gateway_lambda_proxy_integration_with_is_base_64_encoded
    - [ ]  test_put_integration_dynamodb_proxy_validation_with_request_template
    - [ ]  test_multiple_api_keys_validate
    - [ ]  test_apigateway_with_step_function_integration[StartExecution]
    - [ ]  test_apigw_test_invoke_method_api
    - [ ]  test_apigw_stage_variables[dev]
- [ ]  tests.aws.services.apigateway.test_apigateway_import.TestApiGatewayImportRestApi
    - [ ]  test_import_with_stage_variables

### CODE REVIEW

File: tests/aws/services/apigateway/test_apigateway_basic.py [hunk 1]
```diff
@@ -124,11 +132,6 @@ def integration_lambda(create_lambda_function):
 
 class TestAPIGateway:
     # endpoint paths
-    API_PATH_LAMBDA_PROXY_BACKEND = "/lambda/foo1"
-    API_PATH_LAMBDA_PROXY_BACKEND_WITH_PATH_PARAM = "/lambda/{test_param1}"
-    API_PATH_LAMBDA_PROXY_BACKEND_ANY_METHOD = "/lambda-any-method/foo1"
-    API_PATH_LAMBDA_PROXY_BACKEND_ANY_METHOD_WITH_PATH_PARAM = "/lambda-any-method/{test_param1}"
-    API_PATH_LAMBDA_PROXY_BACKEND_WITH_IS_BASE64 = "/lambda-is-base64/foo1"
```
Comment (viren-nadkarni) [inline tests/aws/services/apigateway/test_apigateway_basic.py:131]: Scope is changed to module-level to allow test parametrisation

File: tests/aws/services/apigateway/test_apigateway_basic.py [hunk 2]
```diff
@@ -400,30 +404,49 @@ def test_invoke_endpoint_cors_headers(
...
+    ):
+        role_arn = create_iam_role_with_policy(
+            RoleName=f"role-apigw-lambda-{short_uid()}",
+            PolicyName=f"policy-apigw-lambda-{short_uid()}",
+            RoleDefinition=APIGATEWAY_ASSUME_ROLE_POLICY,
+            PolicyDefinition=APIGATEWAY_KINESIS_POLICY,
         )
 
-    @markers.aws.unknown
-    def test_api_gateway_lambda_proxy_integration_with_path_param(self, integration_lambda):
```
Comment (viren-nadkarni) [inline tests/aws/services/apigateway/test_apigateway_basic.py:409]: Test is merged and parametrised with the previous one

**Review by bentsku — COMMENTED** [review]
This looks good to me! I'll let @dfangl give the send-off, but it seems it fixes all the tests by properly passing a role, that's great! Thanks for fixing all those tests, can't wait to finally make them AWS validated.

**Review by dfangl — APPROVED** [review]
Being able to use credentials for our integrations is definitely a good idea!

Building on my meeting with sannya last week, I want to add that we should try to get those tests AWS validated, as well as also create some that only use resource based policies, without a role provided.

**Pull request merged**